src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java

```diff
@@ -0,0 +1,129 @@
+package mate.academy.shop.dao.hibernate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.shop.anotation.Dao;
+import mate.academy.shop.anotation.Inject;
+import mate.academy.shop.dao.BucketDao;
+import mate.academy.shop.dao.ItemDao;
+import mate.academy.shop.dao.UserDao;
+import mate.academy.shop.factory.HibernateUtil;
+import mate.academy.shop.model.Bucket;
+import mate.academy.shop.model.Item;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class BucketDaoHibernateImpl implements BucketDao {
+    @Inject
+    private static ItemDao itemDao;
+    @Inject
+    private static UserDao userDao;
+
+    @Override
+    public Optional<Bucket> create(Bucket bucket) {
+        Long bucketId = null;
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
+            transaction = session.beginTransaction();
+            bucketId = (Long) session.save(bucket);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        bucket.setId(bucketId);
+        return Optional.of(bucket);
+    }
+
+    @Override
+    public Optional<Bucket> get(Long bucketId) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
+            Bucket bucket = session.get(Bucket.class, bucketId);
+            return Optional.of(bucket);
+        }
+    }
+
+    @Override
+    public Optional<Bucket> update(Bucket bucket) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
+            transaction = session.beginTransaction();
+            session.update(bucket);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return Optional.of(bucket);
+    }
+
+    @Override
+    public Optional<Bucket> delete(Long id) {
+        Transaction transaction = null;
+        Bucket bucket = get(id).get();
+        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
+            transaction = session.beginTransaction();
+            session.delete(bucket);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return Optional.of(bucket);
+    }
+
+    @Override
+    public void addBucketForUser(Long userId, Long bucketId) {
```

Comment:
do we need this method?

Reply:
I will delete all of it, when I will clean the project